app/scripts/background.js

```diff
@@ -472,3 +504,26 @@ extension.runtime.onInstalled.addListener(({reason}) => {
     platform.openExtensionInBrowser()
   }
 })
+
+extension.runtime.onMessage.addListener((message, sender, sendResponse) => {
+  if (message.type === 'notifyTabId' && sender.tab) {
+    const originMatch = message.origin && message.origin.match(/\/\/([^/:]+)\/?/)
+    tabIdOriginMap[sender.tab.id] = originMatch && originMatch.length && originMatch[1]
+    sendResponse({ tabId: sender.tab.id })
+  }
+})
+
+extension.tabs.onActivated.addListener(({ tabId }) => {
+  if (previousTabId !== tabId) {
+    Object.keys(openNonMetamaskTabsIDs).forEach(key => {
```

Comment:
there seems to be a bit of code duplication here, not a show stopper, but maybe this could be broken off into a function that gets called instead? looks like you're doing something similar on lines 407 and 393

Reply:
Thanks. Done in 8d82072a9